src/python/pants/backend/native/subsystems/binaries/binutils.py

```diff
@@ -30,7 +30,9 @@ def linker(self):
     return Linker(
       path_entries=self.path_entries(),
       exe_filename='ld',
-      library_dirs=[])
+      library_dirs=[],
```

Comment:
Slightly confused in the distinction between this param and the one below it. This is slightly ambiguous to me as well: 
```  @abstractproperty
  def library_dirs(self):
    """Directories containing shared libraries required for a subprocess to run."""```
Maybe add an abstract property with a docstring to the `Linker` class?

Reply:
This makes things a lot neater! Added `LinkerMixin` in b2a750bf6!